azkaban-common/src/main/java/azkaban/project/NodeBeanLoader.java

```diff
@@ -86,15 +90,88 @@ public AzkabanNode toAzkabanNode(final NodeBean nodeBean) {
     }
   }
 
+  private void validateSchedule(final FlowTriggerBean flowTriggerBean) {
+    final Map<String, String> scheduleMap = flowTriggerBean.getSchedule();
+
+    Preconditions.checkNotNull(scheduleMap, "flow trigger schedule must not be null");
+
+    Preconditions.checkArgument(scheduleMap.containsKey("type") && scheduleMap.get("type")
+        .equals("cron"), "flow trigger schedule type must be cron");
+
+    Preconditions.checkArgument(scheduleMap.containsKey("value") && CronExpression
+        .isValidExpression(scheduleMap.get("value")), "flow trigger schedule value must be a "
+        + "valid cron expression");
+
+    Preconditions.checkArgument(scheduleMap.containsKey("value") && CronExpression
+        .isValidExpression(scheduleMap.get("value")), "flow trigger schedule value must be a "
+        + "valid cron expression");
+
+    Preconditions.checkArgument(scheduleMap.size() == 2, "flow trigger schedule must "
+        + "contain type and value only");
+  }
+
+  private void validateFlowTriggerBean(final FlowTriggerBean flowTriggerBean) {
+    // validate max wait mins
+    Preconditions.checkArgument(flowTriggerBean.getMaxWaitMins() >= 1, "max wait min must be "
+        + "longer than 1 min");
+
+    validateSchedule(flowTriggerBean);
+    validateTriggerDependencies(flowTriggerBean.getTriggerDependencies());
+  }
+
+  /**
+   * check uniqueness of dependency.name
+   */
+  private void validateDepNameUniqueness(final List<TriggerDependencyBean> dependencies) {
+    final Set<String> seen = new HashSet<>();
+    for (final TriggerDependencyBean dep : dependencies) {
+      // set.add() returns false when there exists duplicate
+      Preconditions.checkArgument(seen.add(dep.getName()), String.format("duplicate dependency"
+          + ".name %s found, dependency.name should be unique", dep.getName()));
+    }
+  }
+
+  /**
+   * check uniqueness of dependency type and params
+   */
+  private void validateDepDefinitionUniqueness(final List<TriggerDependencyBean> dependencies) {
+    final Set<String> seen = new HashSet<>();
+    for (final TriggerDependencyBean dep : dependencies) {
+      final Map<String, String> props = dep.getParams();
+      // set.add() returns false when there exists duplicate
+      Preconditions.checkArgument(seen.add(dep.getType() + ":" + props.toString()), String.format
```

Comment:
If params for two deps have the same content but in different order, will the check here fail?

Reply:
yes, since props is a hashmap which is order agnostic.

Reply:
sorry, i took it back, hashmap's printing order is not guaranteed, https://stackoverflow.com/questions/18778410/why-does-tostring-function-of-a-hashmap-prints-itself-with-a-different-order